---
title: "Statistical processing of samples from dada2 output in phyloseq object"
author: "Mike Parr, M.Parr2@ncl.ac.uk"
output:
  pdf_document: 
    toc: yes
    toc_depth: 6
    fig_width: 20
    fig_height: 12
    fig_caption: yes
    number_sections: yes
    keep_tex: yes
  word_document:
    toc: yes
  html_document:
    toc: yes
    theme: united
    toc_depth: 5
    fig_width: 8
    fig_height: 8
    fig_caption: yes
    number_sections: yes
    keep_md: yes
    df_print: kable
---

```{r setup, include=FALSE} 
#R Notebook setup

library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE)
knitr::opts_chunk$set(warning = FALSE, message = FALSE, dev = "png", dpi = 300, cache = TRUE) 
```

# Version Control
```{r echo=FALSE, cache=FALSE}
timevari = Sys.Date()
print(paste0("This file was created on ", timevari, " using ", R.version$version.string))
```

The general code in this document was taken from Callahan et al (2017) and has been modified for the use of bulk soil eDNA sampled from Nafferton Farm, and sequenced at FERA Ltd., York.

# Loading Libraries

Install the necessary packages, you may already have some of these installed.
```{r install packages, eval=FALSE, message=TRUE}
install.packages("BiocManager", dependencies = T)
library(BiocManager)
BiocManager::install("phyloseq", dependencies = T)
BiocManager::install("dada2", dependencies = T)
BiocManager::install("DECIPHER", dependencies = T)
BiocManager::install("S4Vectors", dependencies = T)#
BiocManager::install("ShortRead", dependencies = T)
install.packages("phytools", dependencies = T)
install.packages("phangorn", dependencies = T)
install.packages("ape", dependencies = T)
install.packages("ggplot2", dependencies = T)
install.packages("doParallel", dependencies = T)
```


library(BiocStyle)#library(gridExtra)

Load in the libraries
```{r load libraries, message=FALSE, warning=FALSE, include=FALSE, results='hide'}
library(devtools)   #better base R functions
library(ShortRead)
library(doParallel) #allows for better utilisation of multiple CPU cores
library(DECIPHER)   #DNA FASTA sequence handling
library(dada2)      #infer samples from amplicons 
library(ape)        #for phylogenetic trees
library(phangorn)   #for phylogenetic trees
library(geiger)     #for phylogenetic trees
library(phyloseq)   #packages your microbiome into an easy to use object
library(phytools)   #extra functions for phyloseq packages
library(ggplot2)    #pretty plots
library(microbiome) #for some extra tools for analysing phyloseq packages and microbiomes
#Setting up parallel processing in Windows ----
cl <- makeCluster(detectCores(), type= 'FORK') #Change to FORK if working on linux, PSOCK on windows 
registerDoParallel(cl) #if this doesn't work on your computer due to admin rights, it is fine to exclude these lines. It will be a bit slower though. 
```

Download important files and import your DNA sequences
```{r}
#Download sequences from Github

folder <- "~/R/TestFolder/" #Choose where to create the folder for this lesson

if (file.exists(folder)){
  cat("The folder already exists")
  } else {
    dir.create(folder)
  }

DwnTarget = paste0(folder,"MicroZip.zip")
download.file(DwnTarget, url = "https://github.com/mparr2/Lesson/archive/refs/heads/main.zip")
unzip(DwnTarget, exdir = folder)

#Download reference FASTA file
Silva138 = paste0(folder, "Silva138.fasta.gz")
download.file("https://zenodo.org/record/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1", Silva138) #download silva 138.1


#Unite = paste0(folder, "Unite.fasta.gz")
#download.file("https://files.plutof.ut.ee/public/orig/7B/23/7B235835FAF5C85D7B01E40FEF17F687914CB81A182554C5BD95E3168328E604.tgz", Unite) 

#FungiFasta = "~/R/R files/ITS Reads/FERA.ITS.fasta.gz"

#ASV import
miseq_path = paste0(folder, "Lesson-main/Reads") #folder location of all sequences in .fastq format
list.files(miseq_path)

```

Now we should have our forward (R1) and reverse (R2) files 
```{r}
fnFs <- sort(list.files(miseq_path, pattern="_R1_001.fastq")) #forward reads
fnRs <- sort(list.files(miseq_path, pattern="_R2_001.fastq")) #reverse reads

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq, i.e. everything after the first _ will be removed
sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)
# Specify the full path to the fnFs and fnRs
fnFs <- file.path(miseq_path, fnFs)
fnRs <- file.path(miseq_path, fnRs)
fnFs[1:3] #do these look correct? Might be worth double checking if you aren't sure
```


We will plot the quality profiles of our reads. We will expect the forward reads to be better quality than our reverse reads

Forward
```{r}
plotQualityProfile(fnFs[1:2])
```

Reverse. These will most likely be worse than the forward reads so don't get upset if they look a little rubbish. 
```{r}
plotQualityProfile(fnRs[1:2])
```

Now we have looked at our raw sequences, we will now filter and trim them before dereplicating them. This will create our abundances, the number of DNA reads of each unique sequence observed in our samples. 
```{r}
#ASV binning and filtering
filt_path <- file.path(miseq_path, "filtered") # Place filtered files in filtered/ subdirectory
if(!file_test("-d", filt_path)) dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq.gz"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                    compress=TRUE, multithread= TRUE) # On Windows set multithread=FALSE

#out1 <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, minLen = 20,
#                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
#                     compress=TRUE, multithread= TRUE) # On Windows set multithread=FALSE

#out
head(out)

#infer sequence variants and dereplicate
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames



```

We will also try and learn the error rates so that we can differentiate between differences in the order of base pairs that are "real", and differences introduced by our sequencer.
```{r}
errF <- learnErrors(filtFs, multithread=TRUE)
```

```{r}
errR <- learnErrors(filtRs, multithread=TRUE)
```

To make sure that errors have been calculated properly, the dots should follow the fitted line.
```{r}
plotErrors(errF) #Forward errors
plotErrors(errR) #Reverse errors
```

Now we use DADA2 to group our unique sequences into Amplicon Sequence Variants (ASVs), taking into account our calculated error rates. ASV are roughly analogous to species, although there may be multiple ASVs per species due to intra-specific variation on our barcoding gene region. 
```{r}
dadaFs <- dada(derepFs, err=errF, multithread=TRUE,pool = F)
saveRDS(dadaFs, "~/R/R files/2022-ITS/dadaFsP.rds")
dadaRs <- dada(derepRs, err=errR, multithread=TRUE,pool = F)
saveRDS(dadaRs, "~/R/R files/2022-ITS/dadaRsP.rds")
dadaFs[[1]]

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
saveRDS(mergers, "~/R/R files/2022-ITS/mergersP.rds")
```

Now we will produce our sequence table with our abundances. It is good practice to save these as Rdata files so we can skip past the previous, processing heavy steps if something goes wrong later. 
```{r}
seqtabAll <- makeSequenceTable(mergers[!grepl("Mock", names(mergers))])
table(nchar(getSequences(seqtabAll)))
save(seqtabAll, file = "~/R/R files/2022-ITS/seqtabAll.RData")
```

We will also remove all chimeras, hybrid sequences created during amplification stages. If we don't, these will be interpreted as a new species. The odd one or two will not make a difference, but some studies have shown that chimeras can account for up to 22% of total reads. 
```{r}
seqtabNoC <- removeBimeraDenovo(seqtabAll)
save(seqtabNoC, file = "~/R/R files/2022-ITS/seqtabNoC.RData")
```

Now we have a nice, chimera-free abundance table with our ASVs and their sequences. Now we need to assign taxonomic identifications to each. To do this, we will be using the Silva 138 release reference FASTA. Effectively, it is a file filled with identified 16S rRNA DNA sequences that we can use to identify our species with. We are using the slightly smaller reference file for demonstrative purposes, but the larger files will identify more taxa, more accurately but at the cost of time and processing power. 
```{r}
taxTab <- assignTaxonomy(seqtabNoC, refFasta = FungiFasta, multithread=TRUE)
unname(head(taxTab))
save(taxTab, file = "~/R/R files/2022-ITS/taxTab92k.RData")
```

We will also need our phylogenetic tree. To do this we will first align our sequences and then calculate phylogenetic distances de Novo based on the similarity of the 16S rRNA barcode.
```{r}
#Construct Phylogenetic Tree----
seqs <- getSequences(seqtabNoC)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)

phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) # Note, tip order != sequence order
fit = pml(treeNJ, data=phangAlign)
fitGTR <- update(fit, k=4, inv=0.2)
compact.align <- unique(phangAlign)
length(compact.align) 
length(phangAlign) 
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE, rearrangement = "NNI", control = pml.control(trace = 1)) 
save(fitGTR, file = "~/R/R files/2022-ITS/fitGTR.RData")

```










Finally, we can combine our metadata, abundance table, taxonomy table, and our phylogenetic tree into a neat little package called a phyloseq object. This makes manipulation and statistical analysis of our datasets easy and more reproducible.
```{r}
#Combine data into phyloseq object----
samdf = read.table('~/R/R files/2021-11-22 All Metadata ITS.csv',sep=',',header=T,row.names=1) #metadata
samdf = data.frame(samdf) 

load("~/R/R files/2022-ITS/seqtabNoC.RData") #This allows you to skip the previous time consuming steps
load("~/R/R files/2022-ITS/seqtabAll.RData")


mean(rownames(seqtabAll) %in% rownames(samdf)) == T


taxTab[,1] = gsub("NR_[0-9]* ","", taxTab[,1])
taxTab = gsub( "[a-z]__", "", taxTab)
taxTab[1,1]

ps2 <- phyloseq(otu_table(seqtabNoC, taxa_are_rows=FALSE),
               sample_data(samdf), 
               tax_table(taxTab),phy_tree(fitGTR$tree))
summarize_phyloseq(ps2)
View(tax_table(ps2))

ps2 = subset_taxa(ps2, Kingdom=="Fungi")


save(ps2, file = "~/R/R files/2022-ITS/phyloseq.RData")


```








https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html


